pandas/core/arrays/period.py

```diff
@@ -461,14 +780,139 @@ def _maybe_convert_timedelta(self, other):
         raise IncompatibleFrequency(msg.format(cls=type(self).__name__,
                                                freqstr=self.freqstr))
 
+    # ------------------------------------------------------------------
+    # Formatting
+    def _format_native_types(self, na_rep=u'NaT', date_format=None,
+                             **kwargs):
+        # TODO(DatetimeArray): remove
+        values = self.astype(object)
+
+        if date_format:
+            formatter = lambda dt: dt.strftime(date_format)
+        else:
+            formatter = lambda dt: u'%s' % dt
+
+        if self.hasnans:
+            mask = self._isnan
+            values[mask] = na_rep
+            imask = ~mask
+            values[imask] = np.array([formatter(dt) for dt
+                                      in values[imask]])
+        else:
+            values = np.array([formatter(dt) for dt in values])
+        return values
+
+    def view(self, dtype=None, type=None):
+        # This is to support things like `.asi8`
+        # PeriodIndex's parent does .values.view('i8').
+        # I don't like adding this,
+        return self._data.view(dtype=dtype)
+
+    def repeat(self, repeats, *args, **kwargs):
```

Comment:
Same question as on other PR: is this needed? Where is it used? (in any case not in this file)

Reply:
I would just add these